### DISCUSSION TITLE
[A/L] (2021/22) Combinatoria - Esercizio 29

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#205] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152254412-e8933d36-4360-4948-a438-8bda9a18ea65.png)

**Comment by Rurik-D** [comment]
## Prendiamo come esempio la seguente data: 03-06-1997

### Abbimo quindi: $p = 1997$ e $q = 6 + 3  = 9$

1. Contiamo i modi di scrivere 1997 come somma di 9 interi non negativi.
<center>
Abbiamo di fronte un chiaro problema di scrittura additiva, di facile risoluzione tramite la formula 
</center>


$$
\begin{aligned}
\binom{m+t-1}{t-1} = \binom{m+t-1}{m}
\end{aligned}
$$

<center>
Abbiamo quindi:
</center>

$$
\begin{aligned}
\binom{1997 + 9 - 1}{1997} = \binom{2005}{1997} = \binom{2005}{8}
\end{aligned}
$$

2. Contiamo i modi di scrivere 9 come somma di 1997 interi non negativi.

<center>
Il problema è simile al precedente solo con i valori scambiati.
</center>

$$
\begin{aligned}
\binom{9 + 1997 - 1}{9} = \binom{2005}{9} = \binom{2005}{1996}
\end{aligned}
$$

3. Contiamo le sequenze ordinate di 3 cifre distinte scelte tra 03061997.

<center>
Siccome 306 (con il primo 0) = 306 (con il secondo 0), è inutile contarlo 2 volte, nella mia solunzione quindi escludo a priori le ripetizioni (siccome non potremo neanche avere soluzioni del tipo "300").
</center>
<center>
Contiamo quindi le sequenze ordinate di 3 cifre distinte scelte tra 036197. 
Abbiamo 6 cifre quindi la solunzione sarà banlmente la seguente:
</center>

$$
\begin{aligned}
\frac{6!}{(6-3)!} = 6 * 5 * 4
\end{aligned}
$$

**Comment by Elia-Belli** [comment]
Confermo la soluzione di @Rurik-D, unico errore nei dati, siccome il testo chiede il mese di nascita +5, nel tuo caso $q = 6+5 = 11$ e non $q = 6 +3 = 9$

**Comment by LuginiAndrea** [comment]
Nel punto 2 parla di soluzioni intere _positive_, il che esclude l'uso delle 0 come valore (tutte le variabili hanno valore $\geq$ 1)
Il problema diventa analogo alla distribuzione di $n$ caramelle a $k$ bambini, con tutti i bambini con almeno 1 caramella.
La risposta è quindi $$\binom{1997 + 9 - 1 - 9}{9-1}$$

**Comment by 18xChris** [comment]
> 

Perché hai scritto 1997 + 9? 6 + 5 sarebbe 11
